### ISSUE TITLE
[Bug]: Instance crashed during a query with inner join

### ISSUE THREAD

**Issue by domingozhang** [matrixorigin/matrixone#1603]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):a08aefcbf08fafde85d98ef10dbd26d588716cbe
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

drop table if exists t1,t2,t3;
create table t1 (libname1 varchar(21) not null primary key, city varchar(20));
create table t2 (isbn2 varchar(21) not null primary key, author varchar(20), title varchar(60));
create table t3 (isbn3 varchar(21) not null, libname3 varchar(21) not null, quantity int);
insert into t2 values ('001','Daffy','A duck''s life');
insert into t2 values ('002','Bugs','A rabbit\'s life');
insert into t2 values ('003','Cowboy','Life on the range');
insert into t2 values ('000','Anonymous','Wanna buy this book?');
insert into t2 values ('004','Best Seller','One Heckuva book');
insert into t2 values ('005','EveryoneBuys','This very book');
insert into t2 values ('006','San Fran','It is a san fran lifestyle');
insert into t2 values ('007','BerkAuthor','Cool.Berkley.the.book');
insert into t3 values('000','New York Public Libra',1);
insert into t3 values('001','New York Public Libra',2);
insert into t3 values('002','New York Public Libra',3);
insert into t3 values('003','New York Public Libra',4);
insert into t3 values('004','New York Public Libra',5);
insert into t3 values('005','New York Public Libra',6);
insert into t3 values('006','San Fransisco Public',5);
insert into t3 values('007','Berkeley Public1',3);
insert into t3 values('007','Berkeley Public2',3);
insert into t3 values('001','NYC Lib',8);
insert into t1 values ('New York Public Libra','New York');
insert into t1 values ('San Fransisco Public','San Fran');
insert into t1 values ('Berkeley Public1','Berkeley');
insert into t1 values ('Berkeley Public2','Berkeley');
insert into t1 values ('NYC Lib','New York');

select isbn2,city,libname1,count(libname1) as a from t3 join t1 on libname1=libname3 join t2 on isbn3=isbn2 group by city,libname1;

goroutine 869 [running]:
github.com/matrixorigin/matrixone/pkg/container/vector.Length(...)
 /mnt/vdb/zhangxiao/matrixone/pkg/container/vector/vector.go:166
github.com/matrixorigin/matrixone/pkg/frontend.getDataFromPipeline({0x23030e0, 0xc04d9feea0}, 0xc0044ec8c0)
 /mnt/vdb/zhangxiao/matrixone/pkg/frontend/mysql_cmd_executor.go:218 +0x381
github.com/matrixorigin/matrixone/pkg/sql/colexec/output.Call(0xc04e6f0280, {0x2215f40, 0xc04e6ee3c0})
 /mnt/vdb/zhangxiao/matrixone/pkg/sql/colexec/output/output.go:45 +0x15e
github.com/matrixorigin/matrixone/pkg/vm.Run({0xc04e6f8060, 0x3, 0x18}, 0x7f3ce25091d8)
 /mnt/vdb/zhangxiao/matrixone/pkg/vm/vm.go:47 +0x9c
github.com/matrixorigin/matrixone/pkg/vm/pipeline.(*Pipeline).RunMerge(0xc04e3bb8e0, 0xc04e6f0280)
 /mnt/vdb/zhangxiao/matrixone/pkg/vm/pipeline/pipeline.go:105 +0x126
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).MergeRun(0xc04e690500, {0x28219a0, 0xc004895730})
 /mnt/vdb/zhangxiao/matrixone/pkg/sql/compile/scope.go:455 +0x4aa
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Exec).Run(0xc04e690100, 0xd4e2b7)
 /mnt/vdb/zhangxiao/matrixone/pkg/sql/compile/exec.go:181 +0x116
github.com/matrixorigin/matrixone/pkg/frontend.(*ComputationWrapperImpl).Run(0xc04d512240, 0x2834f68)
 /mnt/vdb/zhangxiao/matrixone/pkg/frontend/mysql_cmd_executor.go:858 +0x1c
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).doComQuery(0xc04e2f6e10, {0xc003fdd440, 0x82})
 /mnt/vdb/zhangxiao/matrixone/pkg/frontend/mysql_cmd_executor.go:1086 +0x23be
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).ExecRequest(0xc04e2f6e10, 0xc04d9fd638)
 /mnt/vdb/zhangxiao/matrixone/pkg/frontend/mysql_cmd_executor.go:1205 +0x5b3
github.com/matrixorigin/matrixone/pkg/frontend.(*Routine).Loop(0xc04d506ea0)
 /mnt/vdb/zhangxiao/matrixone/pkg/frontend/routine.go:99 +0x2be
created by github.com/matrixorigin/matrixone/pkg/frontend.NewRoutine
 /mnt/vdb/zhangxiao/matrixone/pkg/frontend/routine.go:151 +0x1e7

**Comment by nnsgmsone** [comment]
Because the current inner join is only a half-baked product, it cannot project non-grouping and aggregation attributes,  and this restriction will be released later.properties, and this restriction will be released later.

**Comment by nnsgmsone** [comment]
fix in 0.4.0
